internal/controllers/user_devices_controller.go

```diff
@@ -1856,6 +1856,63 @@ func (udc *UserDevicesController) PostMintDevice(c *fiber.Ctx) error {
 	}, sigBytes)
 }
 
+// GetUserDevicesByIntegration godoc
+// @Description Get user devices using integration id
+// @Tags        user-devices
+// @Produce     json
+// @Accept      json
+// @Param       user_device body controllers.GetUserDevicesByIntegration true "all fields are required"
+// @Security    ApiKeyAuth
+// @Success     200 {object} controllers.GetUserDevicesByIntegrationResponse
+// @Security    BearerAuth
+// @Router      /user/devices/:integrationID [post]
+func (udc *UserDevicesController) GetUserDevicesByIntegration(c *fiber.Ctx) error {
+	ctx := context.Background()
+	integrationID := c.Params("integrationID")
+	reqBody := new(GetUserDevicesByIntegration)
+	if err := c.BodyParser(reqBody); err != nil {
+		return fiber.NewError(fiber.StatusBadRequest, "Couldn't parse request JSON body.")
+	}
+
+	logger := udc.log.With().
+		Str("integrationId", integrationID).
+		Str("route", c.Route().Path).
+		Logger()
+	logger.Info().Msg("Attempting to fetch devices by integrationID")
+
+	deviceInteg, err := udc.DeviceDefSvc.GetIntegrationByID(ctx, integrationID)
```

Comment:
We should have some safeguards against long request
```suggestion
	reqCTX, cancel := context.WithTimeout(ctx, time.Second*30)
	defer cancel()
	deviceInteg, err := udc.DeviceDefSvc.GetIntegrationByID(reqCTX, integrationID)
```